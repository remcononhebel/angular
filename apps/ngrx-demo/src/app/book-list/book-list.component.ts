import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Book } from '../model/book.model';

@Component({
  selector: 'ngrx-demo-book-list',
  templateUrl: './book-list.component.html',
  styleUrls: ['./book-list.component.scss'],
})
export class BookListComponent {
  @Input() books: Book[] = [];
  @Output() add = new EventEmitter<string>();
}
