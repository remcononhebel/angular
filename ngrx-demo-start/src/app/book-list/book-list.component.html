<div class="book-list">
  <div class="book-list__item" *ngFor="let book of books">
    <h2 class="book-list__item-title">
      {{ book.volumeInfo.title }}
    </h2>
    <div class="book-list__item-authors">by {{ book.volumeInfo.authors }}</div>
    <div class="book-list__item-add">
      <button (click)="add.emit(book.id)" data-test="add-button">
        Add to Collection
      </button>
    </div>
  </div>
</div>
