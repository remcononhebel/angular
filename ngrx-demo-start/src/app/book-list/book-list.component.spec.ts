import { Spectator, createComponentFactory } from '@ngneat/spectator';
import { BookListComponent } from './book-list.component';
import { ComponentFixture } from '@angular/core/testing';

describe('BookListComponent', () => {
  let spectator: Spectator<BookListComponent>;
  let component: BookListComponent;
  let fixture: ComponentFixture<BookListComponent>;

  const createComponent = createComponentFactory(BookListComponent);

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
    fixture = spectator.fixture;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
