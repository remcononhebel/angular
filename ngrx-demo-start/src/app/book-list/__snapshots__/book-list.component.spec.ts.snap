// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListComponent should match snapshot 1`] = `
<app-book-list
  add={[Function EventEmitter_]}
  books={[Function Array]}
>
  <div
    class="book-list"
  >
  </div>
</app-book-list>
`;
